mmdet/datasets/coco_panoptic.py

```diff
@@ -0,0 +1,543 @@
+        'banner', 'blanket', 'bridge', 'cardboard', 'counter', 'curtain',
+        'door-stuff', 'floor-wood', 'flower', 'fruit', 'gravel', 'house',
+        'light', 'mirror-stuff', 'net', 'pillow', 'platform', 'playingfield',
+        'railroad', 'river', 'road', 'roof', 'sand', 'sea', 'shelf', 'snow',
+        'stairs', 'tent', 'towel', 'wall-brick', 'wall-stone', 'wall-tile',
+        'wall-wood', 'water-other', 'window-blind', 'window-other',
+        'tree-merged', 'fence-merged', 'ceiling-merged', 'sky-other-merged',
+        'cabinet-merged', 'table-merged', 'floor-other-merged',
+        'pavement-merged', 'mountain-merged', 'grass-merged', 'dirt-merged',
+        'paper-merged', 'food-other-merged', 'building-other-merged',
+        'rock-merged', 'wall-other-merged', 'rug-merged'
+    ]
+
+    def __init__(self, formator='{:012}.png', write_png=False, **kwargs):
+        self.formator = formator
+        self.write_png = write_png
+        super(CocoPanoptic, self).__init__(**kwargs)
+
+    def load_annotations(self, ann_file):
+        """Load annotation from COCO Panoptic style annotation file.
+
+        Args:
+            ann_file (str): Path of annotation file.
+
+        Returns:
+            list[dict]: Annotation info from COCO api.
+        """
+        self.coco = COCOPanoptic(ann_file)
+        self.cat_ids = self.coco.get_cat_ids()
+        self.cat2label = {cat_id: i for i, cat_id in enumerate(self.cat_ids)}
+        self.categories = self.coco.cats
+        self.img_ids = sorted(self.coco.get_img_ids())
+        data_infos = []
+        for i in self.img_ids:
+            info = self.coco.load_imgs([i])[0]
+            info['filename'] = info['file_name']
+            data_infos.append(info)
+        return data_infos
+
+    def get_ann_info(self, idx):
+        """Get COCO annotation by index.
+
+        Args:
+            idx (int): Index of data.
+
+        Returns:
+            dict: Annotation info of specified index.
+        """
+        img_id = self.data_infos[idx]['id']
+        ann_ids = self.coco.get_ann_ids(img_ids=[img_id])
+        ann_info = self.coco.load_anns(ann_ids)
+        # filter out unmatched images
+        ann_info = [i for i in ann_info if i['image_id'] == img_id]
+        return self._parse_ann_info(img_id, ann_info)
+
+    def _parse_ann_info(self, img_id, ann_info):
+        """Parse annotations and load panoptic gts.
+
+        Args:
+            ann_info (list[dict]): Annotation info of an image.
+            img_id (int): The index of an image.
+
+        Returns:
+            dict: A dict containing the following keys: bboxes, bboxes_ignore,
+                labels, masks, seg_map.
+        """
+        gt_bboxes = []
+        gt_labels = []
+        gt_bboxes_ignore = []
+        gt_mask_infos = []
+
+        for i, ann in enumerate(ann_info):
+            mask_info = dict()
+
+            category_id = ann['category_id']
+            contiguous_cid = self.cat2label[category_id]
```

Comment:
contiguous_cid? contiguous_id?

Reply:
`contiguous_cid` means contiguous_category_id.
Maybe I can change it to `contiguous_cat_id`.